test/crd.go

```diff
@@ -253,3 +255,49 @@ func getBuildOutputFromVolume(logger *logging.BaseLogger, c *clients, namespace,
 	_, err = io.Copy(out, readCloser)
 	return buf.String(), nil
 }
+
+func createPipelineRunServiceAccount(namespace string) *corev1.ServiceAccount {
+	return &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      hwSA,
+		},
+		Secrets: []corev1.ObjectReference{{
+			Name: hwSecret,
+		}},
+	}
+}
+
+func createPipelineRunSecret(namespace string) *corev1.Secret {
+	// This service account is JUST a storage reader on gcr.io/build-crd-testing
+	dockerMap := `{
+		    "type": "service_account",
+			"project_id": "build-crd-testing",
+			"private_key_id": "0502a41a812fb64ce56a68ec5832ab0ba11c11e6",
+			"private_key": "-----BEGIN PRIVATE KEY-----\nMIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQC9X4EYOARbxQM8\nD2xXcaZTk+gY8eju98tLqCQqTrGMW9ReT2xOY5AygalPP+p7yXEckwBD/HhM0glI\n7MTLdeeKWr+rA1LwHZyWFW7H0Onf7wnYHEHLWUmc3BCOREDtHRZ7ZrPBf1HTAA/3\nMnW5lZHSN9ojzSSFw6AVu6j6axbBIIJ754LrgKeAYurwfIQ2RLTu201k2IqLYbhb\n3mSVG5R+tbKz1CvM53nHCb7CfuVeWsrA8kk7xHrrLQKMmI9v2sgRugyMAzwz/6zN\nh5/iMxxgeq5W8xkVx3J2nY8JJdHaf/T6AGsONEo43pxeiQVjnRf/Kn10TCc2EsIY\nS49UsZ7BAgMBAAECggEAAupldukCQQuD5U/gamHt7GgW3AMV18eqnHnCa2jlah+S\nAeUGnhgJjNvE+pMFlSv5uf2p2K4ed/oxD6+Cp9ZXDRjgvftIyqjlzbwvFcgzwNuD\nygUkup7HecDsD8TteAobT/Vpwq6zKMrBwCvNkvy6bVloEj5x3bXsaxe95DO/ypu6\nw0W97zxwwDJY6KQcIWMjhrGxvwX7niUCeM4leWBDyGtw1zeJn4hEc6N3ajQacXKc\n+4QlxciamYqQWbPntxWQhhQzcHWLi2l9cFbZD7+uJLF4iN6y8mVNU3K3K1bRYrSD\nRUp3aUVBXmFg+Z/2puVL+mU3j3LLWYyBOkevuOmdgQKBgQDe3GIQkyWIS14TdMOS\nBiKBCDy8h96ehL0HkDbyOkStPKdF9puExZxhy7ojHCIM5FVrpRN25p4sEzwFac+v\nJIFgEvq7mXfmXiXINiePoEQaCnnxDxWgmr0HUKEmS9oMdgLcGU+kCVGNsz7AOum0\nKqY3s22Q9lQ67FOyqiu8WFQ7QQKBgQDZiFhTEZkPEcqZjnwJpTB56ZWP9KTslZP7\nwU4bzi6y+meyf3MJ+4L2SyHc3cpSMbjtNOZCt47b908FUmLXUGNhcwuZjEQxFey0\nm41cS5e4P49b9n6yLBjBrBoqs2WBal2egdhNJSsCWoiZP8/ZT8egXvh7b91jzoK2\nq2PUmA4DgQKBgAWL2IjvEI0Oyx2S11cn/e3WJaTPgNPTG903Upa+qnzhOIx+Maqh\nPF4WsuAy0AogGJwgNJbN8HvKUsETvA5wyyN39XN7w0charFL37osU+WOAzDjnjcs\nAq5O7GPGmXub6EBQBPJhJP1wy4v/K1fHg/F47q4f40LAJOkaYRJD5HzBAoGBANUh\nnIBPJqq4IMvQ6cC9g8B+1xYDea9/Yk1w+SmPGvwrEXy3GKx4K7lKpbPz7m4X33sx\nsEU/+Y2VQmwRa1xQm/53rK7V2l5Jf/D400jRm6ZfSAOvgDTrtZnUGJMrz9E7uNw7\nlgUH3JrivyF/fxMI8qszQbwXP0+xnyqAxEAgduKBAoGAIM+PSNYWCZXxDpSHI18d\njKkoAbw2MoywQIlkeuAn1dXFad1zsXQGdTrmXyv7NPP+8GXBknBLi3cvxTilJISy\nucNrCMiqNASn/dq7cWDFUABgjX16JH2DNFZ/l/UVF3NDAJjXCs1X7yIJyXB6ox/z\naSlqlIMV35DBD7qxRyuKsgk=\n-----END PRIVATE KEY-----\n",
+			"client_email": "pull-secret-testing@build-crd-testing.iam.gserviceaccount.com",
+			"client_id": "107935862033025251352",
+			"auth_uri": "https://accounts.google.com/o/oauth2/auth",
+			"token_uri": "https://accounts.google.com/o/oauth2/token",
+			"auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+			"client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/pull-secret-testing%40build-crd-testing.iam.gserviceaccount.com"
```

Comment:
is this going to make it so that anyone who comes along to this repo and uses this private key can push to ` gcr.io/build-crd-testing`? I think we should avoid commiting this private key if we can :S - maybe there is a way with prow to make a private key available for the test without exposing it anywhere?

Reply:
If we keep this, we should probably add some docs about it to the testing README

Reply:
Could consider declaring encoded base64 variable and decode while creating secret. I will update this